autoload/ale/statusline.vim

```diff
@@ -42,16 +38,13 @@ function! ale#statusline#Update(buffer, loclist) abort
         endif
     endfor
 
-    " Set keys for backwards compatibility.
-    let l:count[0] = l:count.error + l:count.style_error
-    let l:count[1] = l:count.total - l:count[0]
-
     let g:ale_buffer_info[a:buffer].count = l:count
 endfunction
 
-" Get the counts for the buffer, and update the counts if needed.
-function! s:GetCounts(buffer) abort
-    if !exists('g:ale_buffer_info') || !has_key(g:ale_buffer_info, a:buffer)
+" Returns a Dictionary with counts for use in third party integrations.
+function! ale#statusline#Count(buffer) abort
+    " Check if ALE has not run (and return a dummy if so).
+    if !has_key(get(g:, 'ale_buffer_info', {}), a:buffer)
```

Comment:
I'd probably revert your changes to this file for now, but you can keep this `has_key` part.